src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -229,6 +230,25 @@ public string HubName
         // to mock the value from ExtensionConfigContext. It should not be used in production code paths.
         internal Func<Uri> WebhookUriProviderOverride { get; set; }
 
+        internal static void ResolveAppSettingOptions(DurableTaskOptions options, INameResolver nameResolver)
+        {
+            if (options == null)
+            {
+                throw new InvalidOperationException($"{nameof(options)} must be set before resolving app settings.");
```

Comment:
I don't mind making this change but I want to clarify that this method isn't just copied for `DurableTaskExtension.cs`, it's been refactored out of `DurableTaskExtension.cs` so that it can be used _here_ (for TBS purposes) and in the `DurableTaskExtension.cs`. With that in mind, do you still want me to change `InvalidOperationException`, or keep it as is.

I could also just copy the code in case we want different exception types in different locations.